Integrate svcat docs with Service Catalog's

carolynvs opened the pull request:
Fixes #1682.

* Move the contents of the svcat README into the install guide and a new cli doc under /docs.
* Update the walkthrough to use svcat where appropriate.

jberkhahn commented on docs/cli.md:
```diff
@@ -1,70 +1,23 @@
+onto your cluster. If you haven't, please see [install.md](install.md).
 
 ## Plugin
 To use svcat as a plugin, run the following command after downloading:
```
I would maybe say "To use svcat as a kubectl plugin", just so people know what we're talking about, esp. with plugins still an alpha feature.

jberkhahn commented on docs/cli.md:
```diff
@@ -75,9 +28,23 @@ when running in plugin mode, so instead of using `--flag` you must specify a val
+* [View the details of a service instance](#view-the-details-of-a-service-instance)
+* [Unbind all applications from an instance](#unbind-all-applications-from-an-instance)
+* [Unbind a single application from an instance](#unbind-a-single-application-from-an-instance)
+* [Delete a service instance](#delete-a-service-instance)
```
I can't find where the content for this header is, but is this recursive? Does it delete all the bindings for me or do I have to do that manually before deprovisioning the instance?

carolynvs replied:
If you view the md file instead of the diff, you can click on the links. 😀 

https://github.com/carolynvs/service-catalog/blob/svcat-walkthrough2/docs/cli.md#delete-a-service-instance

`svcat deprovision` doesn't delete the bindings for you. There is `svcat unbind INSTANCE_NAME` which does nuke everything for you. It would be nice to add a flag that combined the two, e.g. `svcat deprovision --unbind`. I'm not sure it's a safe default to unbind an instance that's in use, but I could go either way on that.

jberkhahn replied:
I would add a blurb about that in the thing for deleting a service instance then - "be sure to delete all bindings before deprovisioning the instance" or something.

pmorie replied:
I'm going to add a CLI category to our resources so that you'll be able to do `kubectl delete service-catalog-all` and it will get everything in the ns.

carolynvs replied:
That will be so helpful! 💯

jberkhahn commented:
Other than this this LGTM

pmorie commented on docs/walkthrough.md:
````diff
@@ -50,132 +51,92 @@ When we create this `ClusterServiceBroker` resource, the service catalog control
+We can check the status of the broker using `svcat describe`:
 
 ```console
-$ kubectl get clusterservicebrokers ups-broker -o yaml
````
I would like to keep this content in - we should show both kubectl and svcat, and explain where svcat gets you more than just kubectl.

carolynvs replied:
@pmorie  Do you have a suggestion for how to show/mention the two separate commands? People didn't like the "run one of the commands" lead ins for all the code blocks.

Maybe just say at the top that we are following a specific convention and that they can run either command provided? That way we don't have to awkwardly explain the dual commands each time?

arschles replied:
I feel like we should have a separate doc for `kubectl` workflow, and recommend that folks follow the `svcat` flow

carolynvs replied:
If our docs were on a real site that was javascript enabled (like on a githubpages site, instead of github renderings of markdown source) we could display a single code block with tabs for the preferred tool, e.g. 

```
| *svcat* | kubectl |
$ svcat command
svcat output
```

pmorie commented on docs/walkthrough.md:
````diff
@@ -50,132 +51,92 @@ When we create this `ClusterServiceBroker` resource, the service catalog control
+provides. We can view the `ClusterServiceClass` resources available by executing:
 
 ```console
-$ kubectl get clusterserviceclasses -o=custom-columns=NAME:.metadata.name,EXTERNAL\ NAME:.spec.externalName
````
same thing here (and elsewhere), let's keep the kubectl stuff in and add svcat, since users are probably most familiar with that tool in a vacuum.

pmorie commented on docs/walkthrough.md:
```diff
@@ -50,132 +51,92 @@ When we create this `ClusterServiceBroker` resource, the service catalog control
-NAME                                   EXTERNAL NAME
-86064792-7ea2-467b-af93-ac9694d96d52   default
+$ svcat get plans
+   NAME                   CLASS                       DESCRIPTION                         UUID
```
Wondering to myself whether we should even show the UUID column unless a user wants to see it (not feedback on this PR :))

carolynvs replied:
I have wanted to hide UUID since the beginning! :trollface:  I'll submit a PR for that separately.

carolynvs replied:
I was thinking that we wouldn't show UUID in the table formatted output, and only show it for describe. So it's still accessible from svcat, but not one of the "chosen" fields that are printed all the time.

carolynvs replied:
I've started an issue about UUID and provide example output of what I'm proposing.

https://github.com/kubernetes-incubator/service-catalog/issues/1804

carolynvs commented:
I've rebased and incorporated the changes suggested from the review.

Check out https://github.com/carolynvs/service-catalog/blob/svcat-walkthrough2/docs/walkthrough.md#step-2---creating-a-clusterservicebroker-resource to see what the svcat/kubectl command formatting looks like.

pmorie commented:
Thanks a lot for the doc changes :)
